oap-stdlib/src/main/java/oap/util/ConcurrentAssocMap.java

```diff
@@ -0,0 +1,93 @@
+package oap.util;
+
+
+import com.google.common.base.Preconditions;
+import lombok.NoArgsConstructor;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class is a concurrent version of OAP AssocList<K,V>
+ * @param <K>
+ * @param <V>
+ */
+@SuppressWarnings( "AbstractClassName" )
+@NoArgsConstructor
+public abstract class ConcurrentAssocMap<K, V> {
+
```

Comment:
extends AbstractMap<K,V> ?

Reply:
Decided not to extend it and to avoid writing custom serializator for that, to present as list of entres at the end.

Reply:
ok, what about `extends Iterable<V>`?